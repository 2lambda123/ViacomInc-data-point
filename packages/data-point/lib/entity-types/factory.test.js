/* eslint-env jest */

const { createEntityFactory } = require('./factory')

// TODO have a test that actually creates an entity with the result from createEntityFactory

describe('entity-types/factory#createEntityFactory', () => {
  test('should create an entity factory that returns an object with the expected shape', () => {
    const factory = createEntityFactory('Hash')
    expect(factory('nuts', {})).toMatchSnapshot()
  })
})
