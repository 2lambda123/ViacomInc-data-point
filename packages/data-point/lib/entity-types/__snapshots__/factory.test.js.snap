// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entity-types/factory#createEntityFactory should create an entity factory that returns an object with the expected shape 1`] = `
Object {
  "definition": Object {
    "Hash:nuts": Object {},
  },
  "id": "Hash:nuts",
  "name": "nuts",
  "spec": Object {},
  "type": "Hash",
}
`;
